submodules/LIB/hardware/syn/Makefile

```diff
@@ -10,44 +10,39 @@ include ../../config_build.mk
 REMOTE_BUILD_DIR=$(USER)/$(BUILD_DIR_NAME)
 REMOTE_SYN_DIR=$(REMOTE_BUILD_DIR)/hardware/syn
 
-#include the module's headers and sources
-VHDR+=$(wildcard ../src/*.vh) $(wildcard ./src/*.vh)
-VSRC+=$(wildcard ../src/*.v) $(wildcard ./src/*.v)
+OUTPUT_DIR=results
+
+NODE ?= umc130
+UFLAGS+=COV=$(COV)
+UFLAGS+=COV_TEST=$(COV_TEST)
 
-INCLUDE+=.
-INCLUDE+=../src
-INCLUDE+=../common_src
 
 #include local syn segment
 ifneq ($(wildcard syn_build.mk),)
 include syn_build.mk
 endif
 
-OUTPUT_DIR=results
+#include the module's headers and sources
+VHDR+=$(wildcard ../src/*.vh)  
+VSRC+=$(wildcard ../src/*.v) 
 
-#default node
-NODE ?= umc130
+INCLUDE+=.
+INCLUDE+=../src
+INCLUDE+=../common_src
 
-SYN_SERVER=$(CADENCE_SERVER)
-SYN_USER=$(CADENCE_USER)
-SYN_SSH_FLAGS=$(CADENCE_SSH_FLAGS)
-SYN_SCP_FLAGS=$(CADENCE_SCP_FLAGS)
-SYN_SYNC_FLAGS=$(CADENCE_SYNC_FLAGS)
+#include local syn segment
+include $(SYNTHESIZER)/build.mk
 
 synth: $(VHDR) $(VSRC)
-	genus -batch -files build.tcl
-
-
-UFLAGS+=COV=$(COV)
-UFLAGS+=COV_TEST=$(COV_TEST)
+	$(SYNTH_COMMAND) $(SYNTHESIZER)/build.tcl 
```

Comment:
this is not very flexible.

synth: $(VHDR) $(VSRC) synth_tool

then you can flexibly create a target synth_tool in build.mk